File: frontend/src/types/report.ts
export type ReportExportRecord = {
  id: number
  project_id: number
  batch_id?: number | null
  report_type: string
  file_name?: string | null
  file_path?: string | null
  data_date?: string | null
  exported_by?: string | null
  exported_at: string
}

export type ReportType = 'dashboard_excel' | 'weekly_word' | 'weekly_pdf' | 'delay_rectification_excel' | 'rectification_tracking' | 'maintenance_report'

export type ReportConfig = {
  include_advanced_chart_analysis: boolean
  use_ai_weekly_text: boolean
  weekly_delayed_item_limit: number
  weekly_matrix_summary_limit: number
  show_data_quality_section: boolean
  show_rectification_summary: boolean
  default_export_format: 'xlsx' | 'docx'
  file_name_include_project_name: boolean
  file_name_include_data_date: boolean
}

export type ReportPreview = {
  report_type: string
  title: string
  items: Array<{ label: string; value: string | number | boolean | string[] | null }>
}

export type ReportHistoryFilters = {
  reportType?: string | null
  projectName?: string | null
  dateFrom?: string | null
  dateTo?: string | null
  keyword?: string | null
}

export type DelayRectificationFilters = {
  discipline?: string | null
  building?: string | null
  floor?: string | null
  delayLevel?: string | null
  calculationMethod?: string | null
}

File: backend/app/schemas/report.py
from datetime import date, datetime

from pydantic import BaseModel


class ReportConfig(BaseModel):
    include_advanced_chart_analysis: bool = True
    use_ai_weekly_text: bool = False
    weekly_delayed_item_limit: int = 30
    weekly_matrix_summary_limit: int = 10
    show_data_quality_section: bool = True
    show_rectification_summary: bool = True
    default_export_format: str = "xlsx"
    file_name_include_project_name: bool = True
    file_name_include_data_date: bool = True


class ReportPreviewResponse(BaseModel):
    report_type: str
    title: str
    items: list[dict[str, str | int | bool | list[str] | None]]


class ReportExportRead(BaseModel):
    id: int
    project_id: int
    batch_id: int | None = None
    report_type: str
    file_name: str | None = None
    file_path: str | None = None
    data_date: date | None = None
    exported_by: str | None = None
    exported_at: datetime

    model_config = {"from_attributes": True}
